### ISSUE TITLE
Override `transform_table` and `inverse_transform_table` in `TaggedTable`

### ISSUE THREAD

**Issue by zzril** [Safe-DS/Library#389]
### Is your feature request related to a problem?

Once #387 is implemented, override `transform_table` and `inverse_transform_table` in the `TaggedTable` class to return `TaggedTable`s.

Note that `TaggedTable`s can contain feature columns, a target column and columns that are neither features nor target.
Use the mappings added in #387 to determine what new columns are what.

### Desired solution

See above.

**Comment by zzril** [comment]
```py
    def transform_table(self, transformer: TableTransformer) -> TaggedTable:
        """
        Apply a learned transformation onto this table.

        This table is not modified.

        Parameters
        ----------
        transformer : TableTransformer
            The transformer which transforms the given table.

        Returns
        -------
        transformed_table : TaggedTable
            The transformed table.

        Raises
        ------
        TransformerNotFittedError
            If the transformer has not been fitted yet.
        ColunmIsTargetError
            If the transformer tries to remove or replace the target column.

        Examples
        --------
        >>> from safeds.data.tabular.transformation import OneHotEncoder
        >>> from safeds.data.tabular.containers import TaggedTable
        >>> table = TaggedTable({"feat1": ["a", "b", "a"], "feat2": ["a", "b", "d"], "target": [1, 2, 3]},"target")
        >>> table
          feat1 feat2  target
        0     a     a       1
        1     b     b       2
        2     a     d       3
        >>> transformer = OneHotEncoder().fit(table, table.features.column_names)
        >>> table.transform_table(transformer)
           feat1__a  feat1__b  feat2__a  feat2__b  feat2__d  target
        0       1.0       0.0       1.0       0.0       0.0       1
        1       0.0       1.0       0.0       1.0       0.0       2
        2       1.0       0.0       0.0       0.0       1.0       3
        """
        try:
            transformed_table = transformer.transform(self)
        except ColumnIsTargetError as e:  # can happen for example with OneHotEncoder
            raise ColumnIsTargetError(self.target.name) from e  # Re-throw for shorter stacktrace
        # For future transformers, it may also happen that they remove the target column without throwing.
        # If this ever happens, comment-in these lines (currently out-commented b/c of code coverage):
        # if self.target.name in transformer.get_names_of_removed_columns():
        #     raise ColumnIsTargetError(self.target.name)
        return TaggedTable._from_table(transformed_table, self.target.name)

    def inverse_transform_table(self, transformer: InvertibleTableTransformer) -> TaggedTable:
        """
        Invert the transformation applied by the given transformer.

        This table is not modified.

        Parameters
        ----------
        transformer : InvertibleTableTransformer
            The transformer that was used to create this table.

        Returns
        -------
        table : TaggedTable
            The original table.

        Raises
        ------
        TransformerNotFittedError
            If the transformer has not been fitted yet.

        Examples
        --------
        >>> from safeds.data.tabular.transformation import OneHotEncoder
        >>> from safeds.data.tabular.containers import TaggedTable
        >>> table = TaggedTable({"feat1": ["a", "b", "a"], "feat2": ["a", "b", "d"], "target": [1, 2, 3]}, "target")
        >>> table
          feat1 feat2  target
        0     a     a       1
        1     b     b       2
        2     a     d       3
        >>> transformer = OneHotEncoder().fit(table, table.features.column_names)
        >>> transformed_table = table.transform_table(transformer)
        >>> transformed_table
           feat1__a  feat1__b  feat2__a  feat2__b  feat2__d  target
        0       1.0       0.0       1.0       0.0       0.0       1
        1       0.0       1.0       0.0       1.0       0.0       2
        2       1.0       0.0       0.0       0.0       1.0       3
        >>> transformed_table.inverse_transform_table(transformer)
          feat1 feat2  target
        0     a     a       1
        1     b     b       2
        2     a     d       3
        """
        return TaggedTable._from_table(transformer.inverse_transform(self), self.target.name)

```

**Comment by Marsmaennchen221** [comment]
This is the code for the two out-commented lines in `transform_table`:
```py
class TransformerRemoveTargetColumnOnTransform(TableTransformer):

    _removed_columns = []

    def fit(self, table: Table, column_names: list[str] | None) -> TableTransformer:
        return self

    def transform(self, table: Table) -> Table:
        if isinstance(table, TaggedTable):
            self._removed_columns = [table.target.name]
            return table.remove_columns([table.target.name])

    def get_names_of_added_columns(self) -> list[str]:
        pass

    def get_names_of_changed_columns(self) -> list[str]:
        pass

    def get_names_of_removed_columns(self) -> list[str]:
        return self._removed_columns

    def is_fitted(self) -> bool:
        return True


class TestTransformTarget:

    def test_should_raise_if_target_column_gets_removed(self):
        with pytest.raises(ColumnIsTargetError, match=r"Illegal schema modification: Column \"target\" is the target column and cannot be removed."):
            Table({"f1": [1, 2], "target": [8, 9]}).tag_columns("target", ["f1"]).transform_table(TransformerRemoveTargetColumnOnTransform().fit(Table(), []))

```

**Comment by zzril** [comment]
2 old testcases:

`test_transform_table.py`:
```py
import pytest
from safeds.data.tabular.containers import TaggedTable
from safeds.data.tabular.transformation import OneHotEncoder
from safeds.exceptions import ColumnIsTargetError

from tests.helpers import assert_that_tagged_tables_are_equal


def test_should_transform_table() -> None:
    table = TaggedTable({"feat1": ["a", "b", "a"], "feat2": ["a", "b", "d"], "target": [1, 2, 3]}, "target")
    transformer = OneHotEncoder().fit(table, table.features.column_names)
    transformed_table = table.transform_table(transformer)
    expected = TaggedTable(
        {
            "feat1__a": [1.0, 0.0, 1.0],
            "feat1__b": [0.0, 1.0, 0.0],
            "feat2__a": [1.0, 0.0, 0.0],
            "feat2__b": [0.0, 1.0, 0.0],
            "feat2__d": [0.0, 0.0, 1.0],
            "target": [1, 2, 3],
        },
        "target",
    )
    assert_that_tagged_tables_are_equal(transformed_table, expected)


def test_should_raise_column_is_target() -> None:
    table = TaggedTable({"feat1": ["a", "b", "a"], "feat2": ["a", "b", "d"], "target": [1, 2, 3]}, "target")
    transformer = OneHotEncoder().fit(table, None)
    # Passing None means all columns get one-hot-encoded, i.e. also the target column!
    with pytest.raises(
        ColumnIsTargetError,
        match='Illegal schema modification: Column "target" is the target column and cannot be removed.',
    ):
        table.transform_table(transformer)

```

---

`test_inverse_transform_table.py`:
```py
from safeds.data.tabular.containers import TaggedTable
from safeds.data.tabular.transformation import OneHotEncoder

from tests.helpers import assert_that_tagged_tables_are_equal


def test_should_inverse_transform_table() -> None:
    table = TaggedTable({"feat1": ["a", "b", "a"], "feat2": ["a", "b", "d"], "target": [1, 2, 3]}, "target")
    transformer = OneHotEncoder().fit(table, table.features.column_names)
    transformed_table = table.transform_table(transformer)
    expected_transformed = TaggedTable(
        {
            "feat1__a": [1.0, 0.0, 1.0],
            "feat1__b": [0.0, 1.0, 0.0],
            "feat2__a": [1.0, 0.0, 0.0],
            "feat2__b": [0.0, 1.0, 0.0],
            "feat2__d": [0.0, 0.0, 1.0],
            "target": [1, 2, 3],
        },
        "target",
    )
    assert_that_tagged_tables_are_equal(transformed_table, expected_transformed)
    inverse_transformed_table = transformed_table.inverse_transform_table(transformer)
    assert_that_tagged_tables_are_equal(inverse_transformed_table, table)

```

**Comment by lars-reimann** [comment]
No longer needed after #680.
